types/react-bootstrap-table/index.d.ts

```diff
@@ -149,51 +376,119 @@ export interface BootstrapTableProps extends Props<BootstrapTable> {
+	 */
+	footerData?: FooterData[][];
+	/**
+	 * Table footer custom class
+	 */
+	tableFooterClass?: string;
 }
 
-export type SelectRowMode = 'none' | 'radio' | 'checkbox';
+/**
+ * Footer Data for a column.
+ */
+export interface FooterData {
+	/**
+	 * Title to display for the column footer
+	 */
+	label: string;
+	/**
+	 * Index for the column that this footer belongs to
+	 */
+	columnIndex: number;
+	/**
+	 * Text alignment for the data in this footer.
+	 */
+	align?: DataAlignType;
+	/**
+	 * Formatting function for the data in this footer. Used to be able to do things like sum the contents of this
+	 * column in the table so that the footer can be used for totals, etc.
+	 *
+	 * The output value from the formatter function will be used instead of the label if the formatter function is
+	 * defined.
+	 */
+	formatter?(tableData: Array<Row<any>>): string | number | ReactElement<any>;
+}
 
 export interface SelectRow {
 	/**
-	 * For specifing the selection is single(radio) or multiple(checkbox).
+	 * Specify whether the selection column uses single(radio) or multiple(checkbox) selection modes. Required.
 	 */
 	mode: SelectRowMode;
 	/**
-	 * Click the row will trigger selection on that row if enable clickToSelect, default is false.
+	 * If true, clicking the row will trigger selection on that row, default is false.
 	 */
 	clickToSelect?: boolean;
 	/**
-	 * If true, click the row will trigger selection on that row and also trigger cell editing if you enabled cell edit. Default is false.
+	 * If true, clicking the row will trigger selection on that row and also trigger cell editing if you enabled cell edit. Default is false.
 	 */
 	clickToSelectAndEditCell?: boolean;
 	/**
+	 * If true, clicking the row will trigger expanding the row. Default is false.
+	 */
+	clickToExpand?: boolean;
+	/**
 	 * You can assign the background color of row which be selected.
+	 * If your requirement is much complex, you can assign a function to bgColor that
+	 * returns a css color string.
 	 */
-	bgColor?: string;
+	bgColor?: string | ((row: Row<any>, isSelect: boolean) => string);
 	/**
-	 * You can assign the class name of row which be selected.
+	 * You can change the width of the selection column by columnWidth (include units).
 	 */
-	className?: string;
+	columnWidth?: string;
+	/**
+	 * You can assign the class name of selected rows. This can either be a string, or a function that takes two
+	 * arguments: row and isSelect.
+	 *   `row`: The current row data.
+	 *   `isSelect`: Flag indicating whether this particular row is selected.
+	 */
+	className?: string | ((row: Row<any>, isSelect: boolean) => string);
 	/**
 	 * Give an array data to perform which rows you want to be selected when table loading.
-	 * The content of array should be the rowkey which you want to be selected.
+	 * The content of array should be the rowkeys for the rows that you want to be selected.
+	 */
+	selected?: Array<RowKey<any>>;
```

Comment:
This just becomes `any[]`. It does not seem to be an improvement over `string[] | number[]`.

Reply:
Suggestion (not required for approval): If you make `SelectRow` generic (e.g. `interface SelectRow<TRow extends object = any>`), you could be more typesafe here by specifying `selected` as: `selected?: Array<keyof TRow> | number[]`.